cmd/openshift-install/create.go

```diff
@@ -40,10 +40,20 @@ import (
 	"github.com/openshift/library-go/pkg/route/routeapihelpers"
 )
 
+var (
+	createOpts struct {
+		bootstrapTimeout int
+	}
+)
+
 type target struct {
 	name    string
 	command *cobra.Command
 	assets  []asset.WritableAsset
+
+	// commandF can be used to alter the `command` above; this is necessary for
+	// things like persistent flags
+	commandF func(cmd *cobra.Command)
```

Comment:
Not sure why we need this here? It doesn't look like we need this for all commands so why don't we add it only in the newCreateCmd() function?

```

func newCreateCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "create",
		Short: "Create part of an OpenShift cluster",
		RunE: func(cmd *cobra.Command, args []string) error {
			return cmd.Help()
		},
	}
cmd.PersistentFlags().IntVar(&createOpts.bootstrapTimeout, "bootstrap-timeout", 20, "Bootstrap timeout in minutes")
```

Reply:
This is a general solution: each subcommand can now modify the `cobra.Command` instance independently. Setting the bootstrap timeout only makes sense when creating the cluster.

Reply:
Got it my bad. Was hoping to find a way to not add commandF for just one use case but I guess we can't.